### ISSUE TITLE
Captions + Links not working correctly.

### ISSUE THREAD

**Issue by diegoreymendez** [wordpress-mobile/AztecEditor-iOS#893]
## Description

When loading WordPress captions with links, the Example App is failing to parse them correctly.

## Reproduction

Simply insert this (valid caption snippet) into HTML mode and switch back to visual:

```html
[caption]<a href="www.wordpress.com"><img src="image.jpg" height="90" width="135"></a>Some text[/caption]
```

**Expected**

The caption should be parsed correctly into a single image in visual mode.

**Actual**

Some of the caption text is shown as regular text.

**Comment by diegoreymendez** [comment]
Final solution coming up in #910.

**Comment by diegoreymendez** [comment]
Fixed by #910.
